<launch>
    <!-- number of moving objects -->
    <arg name="obj_num" value="10" />
    <arg name="drone_id" value="0"/>

    <arg name="map_size_x" value="50.0"/>
    <arg name="map_size_y" value="25.0"/>
    <arg name="map_size_z" value=" 2.0"/>
    <arg name="odom_topic" value="visual_slam/odom" />
    
    <!-- map -->
    <!-- <node pkg ="map_generator" name ="random_forest" type ="random_forest" output = "screen">
        <param name="map/x_size" value="26" />
        <param name="map/y_size" value="20" />
        <param name="map/z_size" value="3" />
        <param name="map/resolution" value="0.1"/>
        <param name="ObstacleShape/seed" value="1"/>
        <param name="map/obs_num" value="250"/>
        <param name="ObstacleShape/lower_rad" value="0.5"/>
        <param name="ObstacleShape/upper_rad" value="0.7"/>
        <param name="ObstacleShape/lower_hei" value="0.0"/>
        <param name="ObstacleShape/upper_hei" value="3.0"/>
        <param name="map/circle_num" value="250"/>
        <param name="ObstacleShape/radius_l" value="0.7"/>
        <param name="ObstacleShape/radius_h" value="0.5"/>
        <param name="ObstacleShape/z_l" value="0.7"/>
        <param name="ObstacleShape/z_h" value="0.8"/>
        <param name="ObstacleShape/theta" value="0.5"/>
        <param name="pub_rate" value="1.0"/>
        <param name="min_distance" value="0.8"/>
    </node> -->

      <!-- map generator new-->
    <node pkg ="map_generator" name ="random_forest_cxy" type ="random_forest_cxy" output = "screen">    
        <param name="map_generator/x_size"     value="$(arg map_size_x)" />
        <param name="map_generator/y_size"     value="$(arg map_size_y)" />
        <param name="map_generator/z_size"     value="$(arg map_size_z)" />
        <param name="map_generator/resolution" value="0.1"/>        
        
        <param name="map_generator/cylinder_num"    value="40"/>
        <param name="map_generator/sqaure_num" value="40"/>
        <param name="map_generator/cylinder_radius" value="0.2"/>
        <param name="map_generator/cylinder_height" value="3.0"/>
        <param name="map_generator/sqaure_size" value="0.4"/>        

        <param name="map_generator/sensing_range" value="10.0"/>        
        <param name="map_generator/sense_rate" value="10.0"/>        
        <param name="map_generator/min_distance" value="1.5"/>       
    </node>

    <!-- main algorithm params -->
    <include file="$(find ego_planner)/launch/advanced_param.xml">
        <arg name="drone_id" value="$(arg drone_id)"/>
        <arg name="map_size_x_" value="$(arg map_size_x)"/>
        <arg name="map_size_y_" value="$(arg map_size_y)"/>
        <arg name="map_size_z_" value="$(arg map_size_z)"/>
        <arg name="odometry_topic" value="$(arg odom_topic)"/>
        <arg name="obj_num_set" value="$(arg obj_num)" />
        <!-- camera pose: transform of camera frame in the world frame -->
        <!-- depth topic: depth image, 640x480 by default -->
        <!-- don't set cloud_topic if you already set these ones! -->
        <arg name="camera_pose_topic" value="pcl_render_node/camera_pose"/>
        <arg name="depth_topic" value="pcl_render_node/depth"/>
        <!-- topic of point cloud measurement, such as from LIDAR  -->
        <!-- don't set camera pose and depth, if you already set this one! -->
        <arg name="cloud_topic" value="pcl_render_node/cloud"/>
        <!-- intrinsic params of the depth camera -->
        <arg name="cx" value="321.04638671875"/>
        <arg name="cy" value="243.44969177246094"/>
        <arg name="fx" value="387.229248046875"/>
        <arg name="fy" value="387.229248046875"/>
        <!-- maximum velocity and acceleration the drone will reach -->
        <arg name="max_vel" value="2.0" />
        <arg name="max_acc" value="6.0" />
        <!--always set to 1.5 times grater than sensing horizen-->
        <arg name="planning_horizon" value="7.5" />
        <arg name="use_distinctive_trajs" value="true" />
        <!-- 1: use 2D Nav Goal to select goal  -->
        <!-- 2: use global waypoints below  -->
        <arg name="flight_type" value="2" />
        <!-- global waypoints -->
        <!-- It generates a piecewise min-snap traj passing all waypoints -->
        <arg name="point_num" value="4" />
        <arg name="point0_x" value="15" />
        <arg name="point0_y" value="0" />
        <arg name="point0_z" value="1" />
        <arg name="point1_x" value="-15.0" />
        <arg name="point1_y" value="0.0" />
        <arg name="point1_z" value="1.0" />
        <arg name="point2_x" value="15.0" />
        <arg name="point2_y" value="0.0" />
        <arg name="point2_z" value="1.0" />
        <arg name="point3_x" value="-15.0" />
        <arg name="point3_y" value="0.0" />
        <arg name="point3_z" value="1.0" />
        <arg name="point4_x" value="15.0" />
        <arg name="point4_y" value="0.0" />
        <arg name="point4_z" value="1.0" />
    </include>
    <!-- trajectory server -->
    <node pkg="ego_planner" name="drone_$(arg drone_id)_traj_server" type="traj_server" output="screen">
        <remap from="position_cmd" to="drone_$(arg drone_id)_planning/pos_cmd"/>
        <remap from="~planning/bspline" to="drone_$(arg drone_id)_planning/bspline"/>
        <param name="traj_server/time_forward" value="1.0" type="double"/>
    </node>
    <!-- use simulator -->
    <include file="$(find ego_planner)/launch/simulator.xml">
        <arg name="drone_id" value="$(arg drone_id)"/>
        <arg name="map_size_x_" value="$(arg map_size_x)"/>
        <arg name="map_size_y_" value="$(arg map_size_y)"/>
        <arg name="map_size_z_" value="$(arg map_size_z)"/>
        <arg name="init_x_" value="-15"/>
        <arg name="init_y_" value="0"/>
        <arg name="init_z_" value="0.1"/>
        <arg name="odometry_topic" value="$(arg odom_topic)" />
    </include>
</launch>
